permission/permission.go

```diff
@@ -0,0 +1,863 @@
+	acctChan   chan struct{}
+	mux        sync.Mutex
+}
+
+// converts local permissions data to global permissions config
+func populateConfig(config PermissionLocalConfig) types.PermissionConfig {
+	var permConfig types.PermissionConfig
+	permConfig.UpgrdAddress = common.HexToAddress(config.UpgrdAddress)
+	permConfig.InterfAddress = common.HexToAddress(config.InterfAddress)
+	permConfig.ImplAddress = common.HexToAddress(config.ImplAddress)
+	permConfig.OrgAddress = common.HexToAddress(config.OrgAddress)
+	permConfig.RoleAddress = common.HexToAddress(config.RoleAddress)
+	permConfig.NodeAddress = common.HexToAddress(config.NodeAddress)
+	permConfig.AccountAddress = common.HexToAddress(config.AccountAddress)
+	permConfig.VoterAddress = common.HexToAddress(config.VoterAddress)
+
+	permConfig.NwAdminOrg = config.NwAdminOrg
+	permConfig.NwAdminRole = config.NwAdminRole
+	permConfig.OrgAdminRole = config.OrgAdminRole
+
+	// populate the account list as passed in config
+	for _, val := range config.Accounts {
+		permConfig.Accounts = append(permConfig.Accounts, common.HexToAddress(val))
+	}
+	permConfig.SubOrgBreadth.SetString(config.SubOrgBreadth, 10)
+	permConfig.SubOrgDepth.SetString(config.SubOrgDepth, 10)
+
+	return permConfig
+}
+
+// function reads the permissions config file passed and populates the
+// config structure accordingly
+func ParsePermissionConifg(dir string) (types.PermissionConfig, error) {
+	fileName := "permission-config.json"
+	fullPath := filepath.Join(dir, fileName)
+	if _, err := os.Stat(fullPath); err != nil {
+		log.Warn("permission-config.json file is missing", "err", err)
+		return types.PermissionConfig{}, err
+	}
+
+	blob, err := ioutil.ReadFile(fullPath)
+
+	if err != nil {
+		log.Error("error reading permission-config.json file", err)
+		return types.PermissionConfig{}, err
+	}
+	var permlocConfig PermissionLocalConfig
+	err = json.Unmarshal(blob, &permlocConfig)
+	if err != nil {
+		log.Error("error unmarshalling permission-config.json file", err)
+		return types.PermissionConfig{}, err
+	}
+
+	permConfig := populateConfig(permlocConfig)
+	if len(permConfig.Accounts) == 0 {
+		return types.PermissionConfig{}, errors.New("no accounts given in permission-config.json. Network cannot boot up")
+	}
+
+	if permConfig.SubOrgDepth.Cmp(big.NewInt(0)) == 0 || permConfig.SubOrgBreadth.Cmp(big.NewInt(0)) == 0 {
+		return types.PermissionConfig{}, errors.New("sub org breadth depth not passed in permission-config.json. Network cannot boot up")
+	}
+
+	return permConfig, nil
+}
+
+// for cases where the node is joining an existing network, permission service
+// can be brought up only after block syncing is complete. This function
+// waits for block syncing before the starting permissions
+func waitForSync(e *eth.Ethereum) {
+	for !types.GetSyncStatus() {
+		time.Sleep(10 * time.Millisecond)
+	}
+	for e.Downloader().Synchronising() {
+		time.Sleep(10 * time.Millisecond)
+	}
+}
+
+func StartPermissionService(stack *node.Node) {
```

Comment:
`utils.Fatal()` is best used in the front layer not in the service layer. I would make this method return an error and do `utils.Fatal()` at `cmd/geth/main.go`